lib/commons/color/get-background-color.js

```diff
@@ -272,33 +272,35 @@ function includeMissingElements(elmStack, elm) {
 	for (let candidate in elementMap) {
 		// check that TR or LABEL has paired nodeName from elementMap, but don't expect elm to be that candidate
 		if (tagArray.includes(candidate)) {
-			for (let candidateIndex in elementMap[candidate]) {
-				if (candidate.hasOwnProperty(candidateIndex)) {
-					// look up the tree for a matching candidate
-					let ancestorMatch = axe.commons.dom.findUp(
-						elm,
-						elementMap[candidate][candidateIndex]
+			for (
+				let candidateIndex = 0;
+				candidateIndex < elementMap[candidate].length;
+				candidateIndex++
+			) {
+				// look up the tree for a matching candidate
+				let ancestorMatch = axe.commons.dom.findUp(
+					elm,
+					elementMap[candidate][candidateIndex]
+				);
+				if (ancestorMatch && elmStack.indexOf(ancestorMatch) === -1) {
+					// found an ancestor not in elmStack, and it overlaps
+					let overlaps = axe.commons.dom.visuallyOverlaps(
+						elm.getBoundingClientRect(),
+						ancestorMatch
 					);
-					if (ancestorMatch && elmStack.indexOf(ancestorMatch) === -1) {
-						// found an ancestor not in elmStack, and it overlaps
-						let overlaps = axe.commons.dom.visuallyOverlaps(
-							elm.getBoundingClientRect(),
-							ancestorMatch
-						);
-						if (overlaps) {
-							// if target is in the elementMap, use its position.
-							bgNodes.splice(tagArray.indexOf(candidate) + 1, 0, ancestorMatch);
-						}
-					}
-					// nodeName matches value
-					// (such as LABEL, when matching itself. It should be in the list, but Phantom skips it)
-					if (
-						elm.nodeName === elementMap[candidate][candidateIndex] &&
-						tagArray.indexOf(elm.nodeName) === -1
-					) {
-						bgNodes.splice(tagArray.indexOf(candidate) + 1, 0, elm);
+					if (overlaps) {
+						// if target is in the elementMap, use its position.
+						bgNodes.splice(tagArray.indexOf(candidate) + 1, 0, ancestorMatch);
 					}
 				}
+				// nodeName matches value
+				// (such as LABEL, when matching itself. It should be in the list, but Phantom skips it)
+				if (
+					elm.nodeName === elementMap[candidate][candidateIndex] &&
+					tagArray.indexOf(elm.nodeName) === -1
```

Comment:
Scope creep: Need to make sure that these are case insensitive. If these are XHTML nodes, they are going to have a lower case name, whereas on HTML 5 they have upper case.